src/ert/storage/local_experiment.py

```diff
@@ -200,6 +200,26 @@ def ensembles(self) -> Generator[LocalEnsemble, None, None]:
             ens for ens in self._storage.ensembles if ens.experiment_id == self.id
         )
 
+    def get_ensemble_by_name(self, name: str) -> LocalEnsemble:
+        """
+        Retrieves an ensemble by name.
+
+        Parameters
+        ----------
+        name : str
+            The name of the ensemble to retrieve.
+
+        Returns
+        -------
+        local_ensemble : LocalEnsemble
+            The ensemble associated with the given name.
+        """
+
+        for ens in self.ensembles:
+            if ens.name == name:
+                return ens
+        raise KeyError(f"Ensemble with name '{name}' not found")
```

Comment:
Not sure if this should raise in just a "get" function. I'd say get could return `None` and that could be handled by raise in further "stage" if needed.

Reply:
This is an API that can be used directly in for example jupyter notebooks, so it's essential that it raises understandable errors.

Reply:
Sure. I opted for two tests for readability.